### DISCUSSION TITLE
Simple Onedrive log viewer - my solution

### DISCUSSION THREAD

**Discussion by nickzeff** [abraunegg/onedrive#2090] [Show and tell]
I wanted a way to trigger a simple window to show me the latest entries in the onedrive activity log. I used the terminal for a while, but wanted something a bit more elegant. I wrote a script as follows to bring it up in a proper gtk window:

```
#! /bin/bash

PARSER='OFS="\n" {print $1 " " $2, $3, $4, substr($5,0,index($5,":")-1), substr($0,index($0,$6))}'

journalctl --user-unit=onedrive -f --lines=50 | awk -W interactive "$PARSER" | \
yad --title="One Drive Log viewer" --window-icon=logviewer \
    --button=gtk-close --geometry 1200x450-200+50 \
    --list --text="Contents of OneDrive Log" --tail --no-click --no-selection \
    --column Date --column Time --column Host \
    --column Tag --column Message:TIP

exit $?
```
Requires package yad to be installed, obviously. Looks like this:

![image](https://user-images.githubusercontent.com/111174337/184475530-ffacf6d7-9125-4f1c-bdb3-56b48a60e6d4.png)

I'm running Ubuntu 22.04, and I trigger this script from one of the options in a menu on my top panel - I use the excellent Guillotine gnome extension for this (https://extensions.gnome.org//extension/3981/guillotine/) so it looks like the below...

![image](https://user-images.githubusercontent.com/111174337/184475461-0bdfa581-edb1-4336-a6dc-8283a3e750cb.png)

Hope that helps spark off some ideas for someone!

**Comment by abraunegg** [comment]
@nickzeff 
Many thanks for your contribution :) Greatly appreciated :)

**Comment by nickzeff** [comment]
I also tinkered around with some more of Guillotine's options, so I can also start and stop the Onedrive service from the menu as follows

![image](https://user-images.githubusercontent.com/111174337/184491151-f501a4d9-57a9-4e12-8e0e-6e8e4d25064e.png)

This is not that difficult, so happy to share how I did that if anyone is interested.
